kernel/src/table_configuration.rs

```diff
@@ -208,27 +208,31 @@ impl TableConfiguration {
     /// [`TableChanges`]: crate::table_changes::TableChanges
     #[internal_api]
     pub(crate) fn is_cdf_read_supported(&self) -> bool {
+        // TODO: we should probably expand this to ~all supported reader features instead of gating
+        // on a subset here. missing ones are:
+        // - timestampNtz
+        // - v2Checkpoint
+        // - vacuumProtocolCheck
+        // - variantType
         static CDF_SUPPORTED_READER_FEATURES: LazyLock<Vec<TableFeature>> =
-            LazyLock::new(|| vec![TableFeature::DeletionVectors]);
+            LazyLock::new(|| vec![TableFeature::DeletionVectors, TableFeature::ColumnMapping]);
         let protocol_supported = match self.protocol.reader_features() {
             // if min_reader_version = 3 and all reader features are subset of supported => OK
             Some(reader_features) if self.protocol.min_reader_version() == 3 => {
                 ensure_supported_features(reader_features, &CDF_SUPPORTED_READER_FEATURES).is_ok()
             }
-            // if min_reader_version = 1 and there are no reader features => OK
-            None => self.protocol.min_reader_version() == 1,
+            // if min_reader_version = 1 or 2 and there are no reader features => OK
+            None => {
+                self.protocol.min_reader_version() == 1 || self.protocol.min_reader_version() == 2
```

Comment:
Surprised clippy didn't flag this:

```suggestion
                (1..=2).contains(self.protocol.min_reader_version())
```